packages/network/src/connection/NodeWebRtcConnection.ts

```diff
@@ -0,0 +1,318 @@
+    open: () => void
+    message: (msg: string)  => void
+    close: (err?: Error) => void
+    error: (err: Error) => void
+    bufferLow: () => void
+    bufferHigh: () => void
+}
+
+// reminder: only use Connection emitter for external handlers
+// to make it safe for consumers to call removeAllListeners
+// i.e. no this.on('event')
+export const ConnectionEmitter = EventEmitter as { new(): StrictEventEmitter<EventEmitter, Events> }
+
+export class NodeWebRtcConnection extends WebRtcConnection {
+
+    private connection: PeerConnection | null
+    private dataChannel: DataChannel | null
+    private dataChannelEmitter?: EventEmitter
+    private connectionEmitter?: EventEmitter
+    
+    constructor(opts: ConstructorOptions) {
+        super(opts)
+
+        this.connection = null
+        this.dataChannel = null
+        this.onStateChange = this.onStateChange.bind(this)
+        this.onLocalCandidate = this.onLocalCandidate.bind(this)
+        this.onLocalDescription = this.onLocalDescription.bind(this)
+        this.onStateChange = this.onStateChange.bind(this)
+        this.onGatheringStateChange = this.onGatheringStateChange.bind(this)
+        this.onDataChannel = this.onDataChannel.bind(this)
+        
+    }
+
+    protected doSendMessage(message: string): boolean {
+        return this.dataChannel!.sendMessage(message)
+    }
+
+    connect(): void {
+        if (this.isFinished) {
+            throw new Error('Connection already closed.')
+        }
+
+        this.connection = new nodeDataChannel.PeerConnection(this.selfId, {
+            iceServers: this.stunUrls,
+            maxMessageSize: this.maxMessageSize
+        })
+
+        this.connectionEmitter = PeerConnectionEmitter(this.connection)
+
+        this.connectionEmitter.on('stateChange', this.onStateChange)
+        this.connectionEmitter.on('gatheringStateChange', this.onGatheringStateChange)
+        this.connectionEmitter.on('localDescription', this.onLocalDescription)
+        this.connectionEmitter.on('localCandidate', this.onLocalCandidate)
+
+        if (this.isOffering()) {
+            const dataChannel = this.connection.createDataChannel('streamrDataChannel')
+            this.setupDataChannel(dataChannel)
+        } else {
+            this.connectionEmitter.on('dataChannel', this.onDataChannel)
+        }
+
+        this.connectionTimeoutRef = setTimeout(() => {
+            if (this.isFinished) { return }
+            this.logger.warn(`connection timed out after ${this.newConnectionTimeout}ms`)
+            this.close(new Error(`timed out after ${this.newConnectionTimeout}ms`))
+        }, this.newConnectionTimeout)
+    }
+
+    setRemoteDescription(description: string, type: DescriptionType): void {
+        if (this.connection) {
+            try {
+                this.connection.setRemoteDescription(description, type)
+            } catch (err) {
+                this.logger.warn('setRemoteDescription failed, reason: %s', err)
```

Comment:
I can't use the same variable name `logger` in the subclass even though the variable with the same name is private in the base class. So just gonna skip this, because it feels funny to have to name the logger in the subclass as something else 🤷